src/lib/pubkey/kyber/kyber_common/kyber.cpp

```diff
@@ -252,6 +262,27 @@ secure_vector<uint8_t> Kyber_PrivateKey::private_key_bits() const {
    return m_private->mode().keypair_codec().encode_keypair({m_public, m_private});
 }
 
+bool Kyber_PrivateKey::check_key(RandomNumberGenerator& rng, bool strong) const {
+   // As we do not support loading a private key in extended format but rather
+   // always extract it from a 64-byte seed, these checks (as described in
+   // FIPS 203, Section 7.1) should never fail. Particularly, the length checks
+   // and the hash consistency check described in Section 7.2 and 7.3 are
+   // trivial when the private key is always extracted from a seed. The encaps/
+   // decaps roundtrip test is added for completeness.
+
+   if(!Kyber_PublicKey::check_key(rng, strong)) {
+      return false;
+   }
+
+   PK_KEM_Encryptor enc(*this, "Raw");
+   PK_KEM_Decryptor dec(*this, rng, "Raw");
+
+   const auto [c, K] = KEM_Encapsulation::destructure(enc.encrypt(rng));
+   const auto K_prime = dec.decrypt(c);
+
+   return K == K_prime;
```

Comment:
You want to check that the public key is consistent with the private key, don't you?  Isn't this only necessary for Kyber R3, since ML-KEM uses seeds?

Reply:
Essentially, that's what FIPS 203 requests us to do to "check the keys". But you are totally right: This is not needed when extracting the private key from the seeds.